src/main/lua/exasol/adapter/databricks/authentication.lua

```diff
@@ -8,24 +8,34 @@ local M = {}
 
 ---@alias TokenProvider fun(): string
 
+---@param user string?
+---@param password string?
+---@return string header_value
+local function basic_auth_header(user, password)
+    return "Basic " .. base64.encode(string.format("%s:%s", user, password))
+end
+
+---Fetch Databricks OAuth token.
+---See https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html#manually-generate-and-use-access-tokens-for-oauth-m2m-authentication
 ---@param connection_details DatabricksConnectionDetails
 ---@return string
 local function fetch_oauth_token(connection_details)
-    log.trace("Fetching new OAuth M2M token")
+    local url = connection_details.url .. "/oidc/v1/token"
+    log.trace("Fetching OAuth M2M token from %q", url)
     local body = http_client.request({
-        url = connection_details.url .. "/oidc/v1/token",
+        url = url,
         method = "POST",
         headers = {
             ["Content-Type"] = "application/x-www-form-urlencoded",
-            Authorization = "Basic "
-                    .. base64.encode(
-                            string.format("%s:%s", connection_details.oauth_client_id,
-                                          connection_details.oauth_client_secret))
+            Authorization = basic_auth_header(connection_details.oauth_client_id, connection_details.oauth_client_secret)
         },
         request_body = "grant_type=client_credentials&scope=all-apis",
         verify_tls_certificate = false
     })
+    ---@type DatabricksTokenResponse
     local data = cjson.decode(body)
+    log.debug("Received token of length %d of type %q with scope %q, expires in %d", #data.access_token,
```

Comment:
```suggestion
    log.info("Received token of length %d of type %q with scope %q, expires in %d", #data.access_token,
```
=> The expiration date of the token could be of quiet some interest.